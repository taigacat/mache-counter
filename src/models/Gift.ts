/**
 * Represents a gift.
 * @interface
 */
export interface Gift {
  name: string;
  count: number;
  sender?: string;
}
