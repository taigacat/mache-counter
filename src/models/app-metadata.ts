/**
 * Mache.tv metadata
 */
export interface AppMetadata {
  userId: string;
  userName: string;
  liveId: string;
}
