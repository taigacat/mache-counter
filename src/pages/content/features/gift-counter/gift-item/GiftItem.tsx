import React from 'react';
import {Gift} from '../../../../../models/Gift';
import styles from './GiftItem.module.scss';

/**
 * Represents a Prop object.
 * @interface
 */
interface Prop {
  gift: Gift;
}

/**
 * GiftItem component.
 *
 * @component
 *
 * @param {Object} props - The component props.
 * @param {Object} props.gift - Information about the gift.
 *
 * @returns {JSX.Element} The rendered GiftItem component.
 */
export const GiftItem: React.FC<Prop> = ({gift}) => {
  return (
    <div className={styles['gift-item']}>
      <div>{gift.name}</div>
      <div>{gift.count}</div>
    </div>
  );
}
