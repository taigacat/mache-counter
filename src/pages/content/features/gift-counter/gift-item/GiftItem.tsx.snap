// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItem snapshot should match snapshot 1`] = `
<div>
  <div
    class="gift-item"
  >
    <div>
      Test Gift
    </div>
    <div>
      4
    </div>
  </div>
</div>
`;
